test/crowdstart.coffee

```diff
@@ -1,5 +1,169 @@
+      firstName: firstName
+      lastName: lastName
+      email: firstName
+      password: goodPass1
+      passwordConfirm: goodPass1
+
+    p.catch (err)->
+      res = client.lastResponse
+      res.status.should.equal 400
+      res.responseText.error.message.should.equal 'Email is not valid'
+      done()
+
+  it 'should enforce required field requirement', (done) ->
+    p = client.create
+      firstName: ''
+      lastName: lastName
+      email: randomEmail()
+      password: goodPass1
+      passwordConfirm: goodPass1
+
+    p.catch (err)->
+      res = client.lastResponse
+      res.status.should.equal 400
+      res.responseText.error.message.should.equal 'First name cannot be blank'
+      done()
+
+  it 'should allow required but optional field requirement', (done) ->
+    p = client.create
+      #firstName: firstName
+      lastName: lastName
+      email: randomEmail()
+      password: goodPass1
+      passwordConfirm: goodPass1
+
+    p.then (res)->
+      res.status.should.equal 200
+      done()
+
+  it 'should enforce password match requirement', (done) ->
+    p = client.create
+      firstName: firstName
+      lastName: lastName
+      email: randomEmail()
+      password: goodPass1
+      passwordConfirm: goodPass2
+
+    p.catch (err)->
+      res = client.lastResponse
+      res.status.should.equal 400
+      res.responseText.error.message.should.equal 'Passwords need to match'
+      done()
+
+  it 'should enforce password min-length requirement', (done) ->
+    p = client.create
+      firstName: firstName
+      lastName: lastName
+      email: randomEmail()
+      password: badPass1
+      passwordConfirm: badPass1
+
+    p.catch (err)->
+      res = client.lastResponse
+      res.status.should.equal 400
+      res.responseText.error.message.should.equal 'Password needs to be atleast 6 characters'
+      done()
+
+describe 'Crowdstart#login', ->
+  # test users are automatically enabled
+  it 'should login valid users', (done) ->
+    client.getToken().should.equal ''
+
+    p = client.login
+      email: email
+      password: goodPass1
+
+    p.then (res)->
+      res.status.should.equal 200
+      client.getToken().should.not.equal ''
+      done()
+
+  it 'should not login non-existant users', (done) ->
+    p = client.login
+      email: randomEmail()
+      password: goodPass1
+
+    p.catch (err)->
+      res = client.lastResponse
+      res.status.should.equal 401
+      res.responseText.error.message.should.equal 'Email or password is incorrect'
+      done()
+
+  it 'should not login non-existant users', (done) ->
+    p = client.login
+      email: email
+      password: goodPass2
+
+    p.catch (err)->
+      res = client.lastResponse
+      res.status.should.equal 401
+      res.responseText.error.message.should.equal 'Email or password is incorrect'
+      done()
+
+describe 'Crowdstart#account', ->
+  it 'should retieve logged in user data', (done)->
+    p = client.account()
```

Comment:
`account()` is pretty ambiguous. I didn't know if we were creating an account or what was really expected to happen. This is the `login()` method right? I think `login()` or `authenticate()` might be more descriptive/clear.